impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -365,7 +390,8 @@ std::shared_ptr<DeviceBuffer> AllocatorVK::OnCreateBuffer(
 
   VmaAllocationCreateInfo allocation_info = {};
   allocation_info.usage = ToVMAMemoryUsage();
-  allocation_info.preferredFlags = ToVKMemoryPropertyFlags(desc.storage_mode);
+  allocation_info.preferredFlags =
```

Comment:
We should probably use required instead of preferred flags now since we are sure the device supports the type.

Reply:
Ditto for the texture allocation too.

Reply:
Yeah can do that. We will also need to cap check for host visible textures as well.

Reply:
I wasn't able to get this working with some quick effort. lets track switching to required flags once we've chased down all the loose ends.